CMakeLists.txt

```diff
@@ -534,6 +534,12 @@ if(WERROR)
   unset(werror_flag)
 endif()
 
+# Prefer Unix-style package components over frameworks on macOS.
+# This improves compatibility with Python version managers.
```

Comment:
I think these should be tested for existence first (in case set by the user), and set to `INTERNAL` ?

Reply:
From CMake [docs](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry):
>  Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default.

It overrides only when `FORCE` is provided or the type is `INTERNAL`, which implies `FORCE`.

Reply:
Right, my comment was about trying to keep them out of the gui menus.

Reply:
Ah, I see. In that case, wouldn't me be easier and consistent with the other code to mark these both variables as [advanced](https://cmake.org/cmake/help/latest/command/mark_as_advanced.html)?

Reply:
Whoops, of course! That's what I meant to suggest, not Internal. Advanced please :)

Reply:
Thanks! Amended.